dbms/src/Dictionaries/DictionaryBlockInputStream.h

```diff
@@ -0,0 +1,290 @@
+#pragma once
+#include <Columns/ColumnVector.h>
+#include <Columns/ColumnString.h>
+#include <Columns/IColumn.h>
+#include <DataStreams/IProfilingBlockInputStream.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Dictionaries/DictionaryBlockInputStreamBase.h>
+#include <Dictionaries/DictionaryStructure.h>
+#include <Dictionaries/IDictionary.h>
+#include <ext/range.hpp>
+#include <common/logger_useful.h>
+#include <Core/Names.h>
+
+namespace DB 
+{
+
+/* 
+ * BlockInputStream implementation for external dictionaries 
+ * read() returns single block consisting of the in-memory contents of the dictionaries
```

Comment:
Single block will be too large for some dictionaries. It's better to generate blocks of max_block_size.